src/control/server/commands.go

```diff
@@ -79,25 +77,16 @@ func (s *ScanStorCmd) Execute(args []string) (errs error) {
 	srv.scm.Discover(resp)
 
 	if resp.Nvmestate.Status != pb.ResponseStatus_CTRL_SUCCESS {
-		fmt.Fprintln(os.Stderr, "nvme scan: "+resp.Nvmestate.Error)
-		isErrored = true
-	} else {
-		common.PrintStructs("NVMe", srv.nvme.controllers)
+		return fmt.Errorf("nvme scan: %s", resp.Nvmestate.Error)
```

Comment:
the reason you don't want to do this, and hence why it looks a bit ugly, is because we want to process SCM even if NVMe fails, with your change SCM results wont't be printed.

Reply:
Ah, understood. Pushed an update which preserves the original behavior while using the new framework.